### PR TITLE
Write a function in views.py to integrate the model with the web tool. #2194

### PR DESCRIPTION
**Pull request by Sarthak5598**

### CODE REVIEW

**Review by Sarthak5598 — COMMENTED** [review]
Not ready for merge now , Just want to keep it open as everyone is working on it , Would like to know everyones opinion

**Comment by Sarthak5598** [comment]
Hey, can anyone help me with this? For some reason, the `AutoLabel` function isn't getting the `bugdescription`. I've console logged everything, and it's correct from the `report.html` side. It's even getting the label from the function (JSON response), but it can't get `bugdescription` from `report.html`.

**Comment by Sarthak5598** [comment]
Works now !!, no need for a review as of now, thanks

**Comment by Sarthak5598** [comment]
The new implementation works as intended for setting the label but the issue is that the bug reports are no longer being posted on the main page as they used to be. It seems that the issue may be related to how the bug reports are handled after the label selection is done. please help

File: blt/urls.py [hunk 1]
```diff
@@ -506,6 +507,7 @@
     path("change_bid_status/", change_bid_status, name="change_bid_status"),
     path("fetch-current-bid/", fetch_current_bid, name="fetch_current_bid"),
     path("Submitpr/", submit_pr, name="submit_pr"),
+    path("AutoLabel/", AutoLabel, name="AutoLabel"),
```
Comment (arkid15r) [inline blt/urls.py:510]: Could you use something like
```suggestion
    path("issue-auto-label/", AutoLabel, name="AutoLabel"),
```

File: website/templates/report.html [hunk 2]
```diff
@@ -310,7 +313,14 @@ <h3 class="text-xl font-bold leading-none text-red-500">Latest Issues</h3>
             </div>
             <div id="bug-description-container"
                  class="w-[100%] flex justify-between md:pr-10 flex-wrap border-b border-gray-900/10 pb-10 my-10">
-                <div class="w-1/2">{% include "includes/md_editor.html" %}</div>
+                <div class="w-1/2">
+                    {% include "includes/md_editor.html" %}
+                    <button id="GenerateLabel"
```
Comment (arkid15r) [inline website/templates/report.html:318]: It looks like id should start from lowercase character.

File: website/views.py [hunk 3]
```diff
@@ -4539,3 +4543,44 @@ def submit_pr(request):
         return render(request, "submit_pr.html")
 
     return render(request, "submit_pr.html")
+
+
+def AutoLabel(request):
+    dotenv_path = Path("blt\.env")
```
Comment (arkid15r) [inline website/views.py:4549]:
```suggestion
    dotenv_path = Path("blt/.env")
```

File: website/views.py [hunk 4]
```diff
@@ -4539,3 +4543,44 @@ def submit_pr(request):
...
+    load_dotenv(dotenv_path=dotenv_path)
+    OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
+    Token_Limit = 1000
+    Token_per_prompt = 70
```
Comment (arkid15r) [inline website/views.py:4553]:
```suggestion
    token_limit = 1000
    token_per_prompt = 70
```

File: website/views.py [hunk 5]
```diff
@@ -4539,3 +4543,44 @@ def submit_pr(request):
...
+            return JsonResponse({"error": "Rate limit exceeded."}, status=429)
+
+        data = json.loads(request.body)
+        bug_description = data.get("BugDescription")
+        template = """
+        Label: {BugDescription}
+        Options: 0.General, 1.Number error, 2.Functional, 3.Performance, 4.Security, 5.Type, 6.Design, 7.Server down
+        Just return the number corresponding to the appropriate option.
+         """
+        print(bug_description)
```
Comment (arkid15r) [inline website/views.py:4569]: Please remove

**Review by arkid15r — CHANGES_REQUESTED** [review]

File: blt/urls.py [hunk 6]
```diff
@@ -506,6 +507,7 @@
     path("change_bid_status/", change_bid_status, name="change_bid_status"),
     path("fetch-current-bid/", fetch_current_bid, name="fetch_current_bid"),
     path("Submitpr/", submit_pr, name="submit_pr"),
+    path("issue-auto-label/", AutoLabel, name="AutoLabel"),
```
Comment (arkid15r) [inline blt/urls.py:510]: Please follow the existing naming templates for function based views next time.

**Review by arkid15r — COMMENTED** [review]
LGTM

**Review by arkid15r — APPROVED** [review]

**Pull request merged**